source/geod24/LocalRest.d

```diff
@@ -40,6 +40,14 @@
     methods, for some defined time or until unblocked.
     See `sleep`, `filter`, and `clearFilter` for more details.
 
+    Shutdown:
+    The control interface has a shutdown method. You may provide a
+    shutdown routine in the call to `shutdown`, it will be called before
+    the node is destroyed. This `shutdown` method should not schedule any
+    tasks / fibers within the node, otherwise it may introduce race conditions.
+    If this routine throws, LocalRest will log the error in the console and
+    proceed with destroying the stack-allocated node.
```

Comment:
I gave a shot at rewriting it:
```suggestion
    Shutdown:
    The control interface has a shutdown method that can be used to terminate
    a node gracefully. When the shutdown request is handled by the node,
    the event loop will exit and the thread will terminate. While the destructor
    of the node will be called, it might not usable for some actions, for example
    because D destructors may not allocate GC memory, or one may want
    to perform some test-specific operations, such a logging some data in case of failure.
    Therefore, you may provide a shutdown routine in the call to `shutdown`.
    It must accept a single argument of the interface type, and will be called
    with the implementation object just before the node is destroyed.
    If this routine throws, LocalRest will log the error in the console and
    proceed with destroying the stack-allocated node.
    Note that control requests are asynchronous, hence requests from the node
    might be processed / send by the node until the request is actually processed.
```

I also think we should add a mention that all control interface methods are asynchronous (in the previous paragraph). I also wanted to focus more on the shutdown method itself than the optional argument, while keeping the limitations that brought us here and you previously documented.